pyroomacoustics/bss/tests/test_sparseauxiva.py

```diff
@@ -0,0 +1,97 @@
+
+import numpy as np
+from scipy.io import wavfile
+import pyroomacoustics as pra
+
+# We use several sound samples for each source to have a length long enough
+
+wav_files = [
+    ['examples/input_samples/cmu_arctic_us_aew_a0001.wav',
+     'examples/input_samples/cmu_arctic_us_aew_a0002.wav',
+     'examples/input_samples/cmu_arctic_us_aew_a0003.wav',],
+    ['examples/input_samples/cmu_arctic_us_axb_a0004.wav',
+     'examples/input_samples/cmu_arctic_us_axb_a0005.wav',
+     'examples/input_samples/cmu_arctic_us_axb_a0006.wav',]
+    ]
+
+
+def test_sparseauxiva():
+
+    signals = [np.concatenate([wavfile.read(f)[1].astype(np.float32, order='C')
+               for f in source_files])
+               for source_files in wav_files]
+
+    # Define an anechoic room envrionment, as well as the microphone array and source locations.
+    # Room dimensions in meters
+    room_dim = [8, 9]
+    # source locations and delays
+    locations = [[2.5, 3], [2.5, 6]]
+    delays = [1., 0.]
+    # create an anechoic room with sources and mics
+    room = pra.ShoeBox(room_dim, fs=16000, max_order=15, absorption=0.35, sigma2_awgn=1e-8)
+
+    # add mic and good source to room
+    # Add silent signals to all sources
+    for sig, d, loc in zip(signals, delays, locations):
+        room.add_source(loc, signal=np.zeros_like(sig), delay=d)
+
+    # add microphone array
+    room.add_microphone_array(pra.MicrophoneArray(np.c_[[6.5, 4.49], [6.5, 4.51]], room.fs))
+
+    # Compute the RIRs as in the Room Impulse Response generation section.
+
+    # compute RIRs
+    room.compute_rir()
+
+    # Record each source separately
+    separate_recordings = []
+    for source, signal in zip(room.sources, signals):
+        source.signal[:] = signal
+        room.simulate()
+        separate_recordings.append(room.mic_array.signals)
+        source.signal[:] = 0.
+    separate_recordings = np.array(separate_recordings)
+
+    # Mix down the recorded signals
+    mics_signals = np.sum(separate_recordings, axis=0)
+
+    # STFT frame length
+    L = 2048
+
+    # Observation vector in the STFT domain
+    X = np.array([pra.stft(ch, L, L, transform=np.fft.rfft, zp_front=L // 2, zp_back=L // 2)
```

Comment:
The new `transform.STFT` can do this more neatly. `pra.stft` is deprecated and will be eliminated in the near future. The analogous method is `pra.transform.analysis`.

Reply:
Thanks, I did not know that. I changed it for the next PR.